flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -454,18 +453,6 @@ private ListShardsResult listShards(String streamName, @Nullable String startSha
 				}
 			}
 		}
-		// Kinesalite (mock implementation of Kinesis) does not correctly exclude shards before
```

Comment:
I don't think we can remove this until the issue in Kinesalite has been actually fixed. Kinesalte is commonly used for testing and this change would be breaking users.

Reply:
Kinesalite has nothing in relation to prod code. And test code shouldn't affect prod code.
Looks like all Flink tests have passed successfully with proposed changes, haven't they? 

BTW, Kinesalite doesn't look like an active anymore. Last commit has been made `mhart committed on Jun 20, 2018`.
There're pull requests from 2016 ongoing. I've proposed a change to fix the issue in Sep, 2018 but with no feedback:
https://github.com/mhart/kinesalite/pull/77 

So, event if it's used, somebody should consider to shift to more reliable implementation.

Reply:
The whole idea it to remove this peace, cause' it conflicts with AWS specs

Reply:
Do you have an alternative in mind that does require access to an AWS account? We use Kinesalite for end-to-end testing and it has been quite helpful.

Reply:
Unfortunately, I don't.
But keeping Kinesalite requires writing the code which doesn't conform to AWS specs. That's the point.
I think it's better to create your own implementation for mocking kinesis (if there're no reliable ones on the market) rather then keeping workaround implementation which is not supported by AWS officially. 
AWS could change inner implementation (e.g. switch from `^shardId-\\d{12}` to something else, this is not a part of public contract) which would break all users' scripts.

Reply:
Testability is important, and Kinesalite is the only practical way for integration / e2e testing without AWS account dependency that I know of. So I would suggest we look for a solution to bring your work to the finish line w/o compromising test support.

I would suggest we give it couple weeks and maybe few pings more for the Kinesalite change to me merged. If not, then let's work on merging that change into the Lyft fork: https://github.com/lyft/kinesalite

Then make sure a container is available with these changes that does not leave users stranded, before updating Flink with this PR.

Reply:
@tweise thx for a feedback
I've opened another PR for lytf fork: https://github.com/lyft/kinesalite/pull/4
And pinged one more time contributors in original repo.

Will keep you posted.

Reply:
@tweise I've updated PR by putting logic to compare shardIds back. The only difference is I compare it lexicographically. I've added [unit test](https://github.com/apache/flink/pull/6760/files#diff-ca964572c13e36273275e10203360684) to prove it works as expected with Kinesis/Kinesalite shard id's format.
I've resolved conflicts as requested as well.

Can you please take a look?